signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java

```diff
@@ -46,16 +48,18 @@ public class SignerLoader {
   private static final Logger LOG = LogManager.getLogger();
   private static final long FILES_PROCESSED_TO_REPORT = 10;
   private static final int MAX_FORK_JOIN_THREADS = 5;
-  private final Set<Path> loadedConfigurationFiles = new HashSet<>();
+
+  private static final Map<Path, FileTime> loadedConfigFiles = new ConcurrentHashMap<>();
```

Comment:
Is there a performance impact from using ConcurrentHashMap? I think I remember that being a reason for refactoring this at some point to avoid use of ConcurrentHashMap

Reply:
We can consider two approaches here. First make SignerLoader a singelton, second use this Map as a static member variable which is thread-safe. 

The main reason to make it concurrent is to cater for "removal" of config file paths. Since `load` can be called via reload endpoint asynchronously, there is a corner case where an entry is removed in one thread while it is being accessed during loading. On the other hand, if we want to keep track of Paths of all the files that we have read, then we can revert to standard HashMap.